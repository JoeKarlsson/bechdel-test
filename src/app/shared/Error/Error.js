import React from 'react';

const Error = () => <div>Im sorry! Please try again.</div>;

export default Error;
