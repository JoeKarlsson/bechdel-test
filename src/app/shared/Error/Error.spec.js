import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Error from './Error';

configure({ adapter: new Adapter() });

describe('Error', () => {
	let wrapper;

	global.requestAnimationFrame = callback => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<Error />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<Error />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
