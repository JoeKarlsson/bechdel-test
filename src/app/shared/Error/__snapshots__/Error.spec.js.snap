// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page rendering match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
